src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,304 @@
+
+
+def _create_shifter_wrapper(job_dir, image_str):
+    """
+    Creates the Shifter wrapper script.
+    """
+
+    # The content of the Shifter wrapper script
+    shifter_wrapper = f'''#!/bin/bash
+
+image={image_str}
+
+if [ $# -lt 1 ]; then
+    echo "Error: Missing command argument."
+    echo "Usage: shifter_wrapper.sh your-command-arguments"
+    exit 1
+fi
+
+command="$@"
+echo "Running shifter --image=$image $command"
+
+cd {job_dir}
+shifter --image=$image $command
+            '''
+
+    wrapper_file = os.path.join(job_dir, 'shifter_wrapper.sh')
+    with open(wrapper_file, "w") as f:
+        f.write(shifter_wrapper)
+
+    os.chmod(wrapper_file, 0o777)
+
+    return wrapper_file
+
+
+def _create_genome_id_file(genome_ids, genome_id_file):
+    """
+    Create a tab-separated values (TSV) file with a list of genome IDs.
+    """
+    with open(genome_id_file, 'w') as f:
+        f.write("genome_id\n")
+        for genome_id in genome_ids:
+            f.write(f"{genome_id}\n")
+
+
+def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper_file, job_dir, root_dir,
+                      threads=256, program_threads=256, source_file_ext='genomic.fna.gz'):
+    """
+    Create task list file (tasks.txt)
+    """
+    genome_ids = [path for path in os.listdir(source_data_dir) if
+                  os.path.isdir(os.path.join(source_data_dir, path))]
+
+    chunk_size = CHUNK_SIZE[tool]
+    genome_ids_chunks = [genome_ids[i: i + chunk_size] for i in range(0, len(genome_ids), chunk_size)]
+
+    vol_mounts = TOOL_VOLUME_MAP.get(tool, {})
+
+    task_list = '#!/usr/bin/env bash\n'
+    for idx, genome_ids_chunk in enumerate(genome_ids_chunks):
+        task_list += wrapper_file
+
+        genome_id_file = os.path.join(job_dir, f'genome_id_{idx}.tsv')
+        _create_genome_id_file(genome_ids_chunk, genome_id_file)
+
+        env_vars = {'TOOLS': tool, 'LOAD_VER': load_ver, 'SOURCE_DATA_DIR': source_data_dir,
+                    'KBASE_COLLECTION': kbase_collection, 'ROOT_DIR': root_dir,
+                    'NODE_ID': f'job_{idx}', 'GENOME_ID_FILE': genome_id_file,
+                    'THREADS': threads, 'PROGRAM_THREADS': program_threads, 'SOURCE_FILE_EXT': source_file_ext}
+
+        for mount_vol, docker_vol in vol_mounts.items():
+            task_list += f''' --volume={mount_vol}:{docker_vol} '''
+
+        for env_var, env_val in env_vars.items():
+            task_list += f'''--env {env_var}={env_val} '''
+
+        task_list += f'''--entrypoint\n'''
+
+    task_list_file = os.path.join(job_dir, 'tasks.txt')
+    with open(task_list_file, "w") as f:
+        f.write(task_list)
+
+    return task_list_file, len(genome_ids_chunks)
+
+
+def _create_batch_script(job_dir, task_list_file, n_jobs):
+    """
+    Create the batch script (submit_taskfarmer.sl) for job submission
+    """
+    batch_script = f'''#!/bin/sh
+
+#SBATCH -N {n_jobs + 1} -c 64
+#SBATCH -q regular
+#SBATCH --time=4:00:00
```

Comment:
I assume you'll adjust this once you have a better idea of how long it takes to run 1000 / 5000 genomes thru gtdbk / checkm.